unitex/src/fr/umlv/unitex/leximir/delas/EditorDelas.java

```diff
@@ -803,21 +833,24 @@ private void jMenuSaveMouseClicked(java.awt.event.MouseEvent evt) {
                     fileData.put(file, tmp);
                 }
             }
+            Encoding e = ConfigManager.getManager().getEncoding(null);
             for (Map.Entry<String, List<String>> data : fileData.entrySet()) {
                 try {
-                    String fn = data.getKey().contains(File.separator) ? new File(data.getKey()).getName() : data.getKey();
-                    try (OutputStreamWriter out = new OutputStreamWriter(new FileOutputStream(DictionaryPath.allDelas + File.separator + fn),Charset.defaultCharset())) {
-                        for (String lines : data.getValue()) {
-                            UnicodeIO.writeString(out, lines);
-                        }
-                    }
+                	String fn = data.getKey();
+                    File dic = new File(workingDirectory + fn);
+                	try (OutputStreamWriter out = e.getOutputStreamWriter(dic)) {
+                		for (String lines : data.getValue()) {
+                			UnicodeIO.writeString(out, lines);
+                		}
+                	}
                 } catch (IOException ex) {
-                    JOptionPane.showMessageDialog(null, "error :" + ex.getMessage());
+                    JOptionPane.showMessageDialog(null, "error :" + ex.getMessage() + "\nNo changes were done.");
+                    return;
                 }
             }
 
             this.setUnsaved(false);
-            JOptionPane.showMessageDialog(null, "Files where saved successfully");
+            JOptionPane.showMessageDialog(null, "File was saved successfully");
```

Comment:
`The file was successfully saved`

Reply:
I changed it, in the "Save as" option it would be :"The file was saved successfully" as there is only one file saved, and in the "Save" option it can be one or more files saved, is it necessary to count the number of files saved to print a different message?

Reply:
Not for now.